ObjcVoiceQuickstart/ViewController.m

```diff
@@ -232,15 +235,42 @@ - (void)credentialsUpdated:(PKPushCredentials *)credentials {
              } else {
                  NSLog(@"Successfully registered for VoIP push notifications.");
                  
+                 // Save the device token after successfully registered.
+                 [[NSUserDefaults standardUserDefaults] setObject:cachedDeviceToken forKey:kCachedDeviceToken];
+                 
                  /*
-                  * Save the device token after successfully registered.
+                  * The twilio registration binding is refreshed for 365 days upon regirstrtion and when a new
+                  * push notification is dispatch.
+                  * Reset the binding date in UserDefaults.
                   */
-                 [[NSUserDefaults standardUserDefaults] setObject:cachedDeviceToken forKey:kCachedDeviceToken];
+                 [[NSUserDefaults standardUserDefaults] setObject:[NSDate date] forKey:kCachedBindingTime];
              }
         }];
     }
 }
 
+/*
+ * The twilio registration binding is refreshed for 365 days on twilio notify when a new
+ * push-notification is disptached. This method checks if binding exists in UserDefaults,
+ * and if 365 days has been passed then the method will return true, else false.
+ */
+- (BOOL)bindingRequired {
```

Comment:
If I'm a user of this API I want to re-register way ahead of a potential expiry so I would actually set this to 6 months. Otherwise, if I'm updated once expired it means that I'm already not reachable at this point.

Reply:
Make sense, changed logic to check for the half of the TTL